service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java

```
@@ -0,0 +1,213 @@
+            userRepository.save(user);
+            eventService.save(OrderHistory.ADD_VIOLATION, currentUser.getRecipientName()
+                + "  " + currentUser.getRecipientSurname(), order);
+            notificationService.notifyAddViolation(order);
+        } else {
+            throw new OrderViolationException(ORDER_ALREADY_HAS_VIOLATION);
+        }
+    }
+
+    private Violation violationBuilder(AddingViolationsToUserDto add, Order order) {
+        return Violation.builder()
+            .violationLevel(ViolationLevel.valueOf(add.getViolationLevel().toUpperCase()))
+            .description(add.getViolationDescription())
+            .violationDate(order.getOrderDate())
+            .order(order)
+            .build();
+    }
+
+    /**
+     * Method returns detailed information about user violation by order id.
+     *
+     * @param orderId of {@link Long} order id;
+     * @return {@link ViolationDetailInfoDto};
+     * @author Rusanovscaia Nadejda
+     */
+
+    @Override
+    @Transactional
+    public Optional<ViolationDetailInfoDto> getViolationDetailsByOrderId(Long orderId) {
+        User user =
+            userRepository.findUserByOrderId(orderId).orElseThrow(() -> new NotFoundException(EMPLOYEE_NOT_FOUND));
+        return violationRepository.findByOrderId(orderId).map(v -> ViolationDetailInfoDto.builder()
+            .orderId(orderId)
+            .userName(user.getRecipientName())
+            .violationLevel(v.getViolationLevel())
+            .description(v.getDescription())
+            .images(v.getImages())
+            .violationDate(v.getViolationDate())
+            .build());
+    }
+
+    @Override
+    @Transactional
+    public void deleteViolation(Long id, String uuid) {
+        User currentUser = userRepository.findUserByUuid(uuid)
+            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+        Optional<Violation> violationOptional = violationRepository.findByOrderId(id);
```

create service method findUserById that throws exception if not found instead of copy-pasting this logic